File: Data Structures & Algorithms/minimum-interval-including-query/submission-1.cpp
class Solution {
public:
    vector<int> minInterval(vector<vector<int>>& intervals, vector<int>& queries) {
        vector<int> sortedQuery = queries;
        sort(sortedQuery.begin(), sortedQuery.end());
        sort(intervals.begin(), intervals.end());
        priority_queue<tuple<int,int>, vector<tuple<int,int>>, greater<tuple<int,int>>> minHeap;
        unordered_map<int,int> res = {};
        int i = 0;
        for (int q = 0; q < sortedQuery.size(); q++) {
            while (i < intervals.size() && intervals[i][0] <= sortedQuery[q]) {
                minHeap.push({intervals[i][1]-intervals[i][0]+1, intervals[i][1]});
                i++;
            }
            while (minHeap.size() > 0 && get<1>(minHeap.top()) < sortedQuery[q]) {
                minHeap.pop();
            }
            if (minHeap.empty()) {
                res[sortedQuery[q]] = -1;
            } else {
                res[sortedQuery[q]] = get<0>(minHeap.top());
            }
        }
        vector<int> answer;
        for (i = 0; i < queries.size(); i++) {
            answer.push_back(res[queries[i]]);
        }
        return answer;
    }
};

File: Data Structures & Algorithms/pow-x-n/submission-1.py
class Solution:
    def myPow(self, x: float, n: int) -> float:
        if x == 0:
            return 0
        if n == 0:
            return 1
        if n == 1:
            return x
        if n == -1:
            return 1/(x)
        if n % 2 == 1:
            return self.myPow(x * x, n // 2) * x
        else:
            return self.myPow(x * x, n // 2)
